### PR TITLE
Fix regression that broke instancing after glTF cleanup

### PR DESCRIPTION
**Pull request by j9liu**
## Description

Addresses a regression from #1836 that broke instancing, due to an accidental overwrite of the instanced component.

## Issue number or link

N/A

## Author checklist

- [X] I have submitted a [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) (only needed once).
- [X] I have done a full self-review of my code.
- ~[ ] I have updated `CHANGES.md` with a short summary of my change (for user-facing changes).~ - broke before release
- ~[ ] I have added or updated unit tests to ensure consistent code coverage as necessary.~ - don't have yet, but would be nice :)
- ~[ ] I have updated the documentation as necessary.~

## Testing plan

Should be able to test this with any instanced tileset from https://github.com/CesiumGS/cesium/tree/main/Apps/SampleData/Cesium3DTiles/Instanced

## Reviewer checklist

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.

As you review, please go through the checklist below:

- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
    - [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
    - [ ] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
- [ ] Review the unit tests and make sure there are no missing tests or edge cases.
- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [ ] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.

### CODE REVIEW

File: Source/CesiumRuntime/Private/CesiumGltfComponent.cpp [hunk 1]
```diff
@@ -3507,7 +3507,7 @@ ConstructedPrimitiveComponent createPrimitiveComponent(
     const TSharedPtr<FCesiumPrimitiveFeatures>& pInstanceFeatures) {
   ConstructedPrimitiveComponent result;
 
-  if (!instanceTransforms.empty()) {
```
Comment (david-lively) [inline Source/CesiumRuntime/Private/CesiumGltfComponent.cpp:3510]: Any particular reason for this change? They're functionally identical. I feel the one on original is slightly more clear.

**Review by david-lively — COMMENTED** [review]
Looks great!

File: Source/CesiumRuntime/Private/CesiumGltfComponent.cpp (hunk 1 above)
Comment (j9liu) [reply Source/CesiumRuntime/Private/CesiumGltfComponent.cpp:3510]: Ah no, this is a personal preference that leaked into here while I was going over stuff. 😅 Sorry about that
